packages/service-clients/end-to-end-tests/azure-client/src/test/presenceTest.spec.ts

```diff
@@ -0,0 +1,175 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+import { fork, ChildProcess } from "node:child_process";
+
+import { AzureClient, type AzureContainerServices } from "@fluidframework/azure-client";
+import { type AzureUser, ScopeType } from "@fluidframework/azure-client/internal";
+import { AttachState } from "@fluidframework/container-definitions";
+import { ConnectionState } from "@fluidframework/container-loader";
+import { ContainerSchema, type IFluidContainer } from "@fluidframework/fluid-static";
+import {
+	acquirePresenceViaDataObject,
+	ExperimentalPresenceManager,
+	type ExperimentalPresenceDO,
+	type IPresence,
+	// type ISessionClient,
+	// eslint-disable-next-line import/no-internal-modules
+} from "@fluidframework/presence/alpha";
+import { timeoutPromise } from "@fluidframework/test-utils/internal";
+
+import { createAzureClient } from "./AzureClientFactory.js";
+import { configProvider } from "./utils.js";
+
+interface MessageFromChild {
+	event: string;
+	sessionId: string;
+}
+
+interface MessageToParent {
+	command: string;
+	containerId: string;
+	user: AzureUser;
+}
+
+describe(`Presence with AzureClient`, () => {
+	const numClients = 5;
+	const children: ChildProcess[] = [];
+	const connectTimeoutMs = 10_000;
+	const initialUser: AzureUser = {
+		id: "test-user-id-1",
+		name: "test-user-name-1",
+	};
+
+	afterEach(async () => {
+		// kill all child processes after each test
+		for (const child of children) {
+			if (!child.killed) {
```

Comment:
It would probably make more sense to reset the array after each test. Something like - 

```ts
let children: ChildProcess[] = [];
afterEach(async () => {
  // kill all child processes after each test
  for (const child of children) {
    child.kill();
  }
  children = [];
}
```